File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number)
{
    int sum = 0;
    for (unsigned int i =0; i < number; i++)
    {
        sum += pow(number-i, 2);
    }
    return sum;
}

unsigned int square_of_sum(unsigned int number)
{
        int sum = 0;
    for (unsigned int i =0; i < number; i++)
    {
        sum += number - i;
    }
    return pow(sum, 2);
}

unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}

File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    const n = number;
    var sum: usize = 0;
    for (0..n + 1) |i| {
        sum = sum + i;
    }
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    const n = number;
    var sum: usize = 0;
    for (0..n + 1) |i| {
        sum = sum + (i * i);
    }
    return sum;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
